src/parser/type_checking.c

```diff
@@ -19718,6 +19719,14 @@ pt_evaluate_function_w_args (PARSER_CONTEXT * parser, FUNC_TYPE fcode, DB_VALUE
       return 0;
     }
 
+  DB_VALUE **args = in_args;
+  JSON_FUNCTION_ARGS_PREPROCESSOR json_args_mapper;
+  if (pt_is_json_function (fcode))
```

Comment:
are all arguments of all json functions supposed to be converted to utf8?

can you look into function signature and maybe set there expected codeset/collation for PT_GENERIC_TYPE_JSON_DOC and/or PT_GENERIC_TYPE_JSON_VAL?